assets/js/modules/analytics/components/dashboard/DashboardAllTrafficWidget/UserDimensionsPieChart.js

```diff
@@ -63,19 +63,38 @@ export default function UserDimensionsPieChart( { dimensionName, sourceLink, loa
 		const { chart, onSelect } = chartData || {};
 		const { slices } = UserDimensionsPieChart.chartOptions;
 
+		if ( chart && report?.[ 0 ]?.data?.rows ) {
+			// If there is a dimension value set but the initialized chart does not have a selection yet,
+			// find the matching row index and initially select it in the chart.
+			if ( dimensionValue && ! chart.getSelection().length ) {
+				const selectedRow = report[ 0 ].data.rows.findIndex( ( row ) => row.dimensions.includes( dimensionValue ) );
+				if ( selectedRow && slices[ selectedRow ]?.color ) {
+					chart.setSelection( [ { row: selectedRow } ] );
+					setValues( FORM_ALL_TRAFFIC_WIDGET, { dimensionColor: slices[ selectedRow ]?.color } );
+				}
+			}
+
+			// If there is no dimension value set but the initialized chart does have a selection,
+			// ensure it is no longer selected in the chart.
+			if ( ! dimensionValue && chart.getSelection().length ) {
+				chart.setSelection( [] );
+				setValues( FORM_ALL_TRAFFIC_WIDGET, { dimensionColor: '' } );
+			}
+		}
+
```

Comment:
@felixarntz this should be moved to its own `useEffect` hook because the `onReady` callback is executed only once per the chart component lifetime.

Reply:
@eugene-manuilov I tried that originally, but it didn't result in the expected behavior, while having it here does. I agree with you that in principle it should be in a `useEffect`, but there are some broader problems with our charts component - the `onReady` event fires several times. Since `useEffect` is decoupled from that, the chart might not be ready, so that didn't work.

Overall, the approach here is not the one we should go with for the long term, but it addresses the user-facing issue for this release. I've opened #2714 to address this in a better way.

Reply:
@felixarntz see #2717 - it works correctly now (at least when you switch date ranges), however, that PR still requires some work. Let me know how it works on your end.